src/cli/commands/mod.rs

```diff
@@ -0,0 +1,75 @@
+//! This mod contains the different command line implementations.
+
+mod server;
+
+use crate::cli::CommandLineHandler;
+use clap::{Parser, Subcommand};
+use std::fmt::Debug;
+
+/// The `cli` method exposed to handle all the cli commands, ideally from main.
+///
+/// Consider using macro_rules! to handle the registration of new CLI handlers.
+///
+/// # Examples
+/// Sample usage:
+/// ```ignore
+/// # to start the daemon with
+/// ipc-client daemon ./examples/sample_config.toml
+/// ```
+///
+/// To register a new command, add the command to
+/// ```ignore
+/// pub async fn cli() {
+///
+///     // ... other code
+///
+///     let r = match &args.command {
+///         // ... other existing commands
+///         Commands::NewCommand => <NewCommand as CommandLineHandler>::handle(n).await,
+///     };
+///
+///     // ... other code
+/// ```
+/// Also add this type to Command enum.
+/// ```ignore
+/// enum Commands {
+///     NewCommand(<NewCommand as CommandLineHandler>::Arguments),
+/// }
+/// ```
+pub async fn cli() {
+    // parse the arguments
+    let args = IPCAgentCliCommands::parse();
+
+    let r = match &args.command {
+        Commands::Daemon(args) => <server::LaunchJsonRPC as CommandLineHandler>::handle(args).await,
```

Comment:
I think I asked you at some point, but is there an alternative to make this a bit less verbose?

Reply:
updated, simplified the notations.